src/main/java/teammates/ui/controller/InstructorFeedbackResultsDownloadAction.java

```diff
@@ -32,25 +34,40 @@ protected ActionResult execute() throws EntityDoesNotExistException {
 
         String fileContent = "";
         String fileName = "";
+
         try {
+            String questionName = "";
+            if (questionNumber != null) {
+                questionName = "_question" + questionNumber;
+            }
+
             if (section == null || "All".equals(section)) {
                 fileContent = logic.getFeedbackSessionResultSummaryAsCsv(
-                        courseId, feedbackSessionName, instructor.email, filterText, isMissingResponsesShown, isStatsShown);
-                fileName = courseId + "_" + feedbackSessionName;
+                        courseId, feedbackSessionName, instructor.email, filterText,
+                        isMissingResponsesShown, isStatsShown, questionId);
+
+                fileName = courseId + "_" + feedbackSessionName + questionName;
                 statusToAdmin = "Summary data for Feedback Session " + feedbackSessionName
                               + " in Course " + courseId + " was downloaded";
             } else {
                 fileContent = logic.getFeedbackSessionResultSummaryInSectionAsCsv(
                         courseId, feedbackSessionName, instructor.email, section,
-                        filterText, isMissingResponsesShown, isStatsShown);
-                fileName = courseId + "_" + feedbackSessionName + "_" + section;
+                        filterText, isMissingResponsesShown, isStatsShown, questionId);
+                fileName = courseId + "_" + feedbackSessionName + "_" + section + questionName;
                 statusToAdmin = "Summary data for Feedback Session " + feedbackSessionName
                               + " in Course " + courseId + " within " + section + " was downloaded";
             }
         } catch (ExceedingRangeException e) {
             // not tested as the test file is not large enough to reach this catch block
+            String smallerDownloadOptionSuggestionMessage = "";
+            if (questionNumber == null) {
+                smallerDownloadOptionSuggestionMessage = "question by clicking the question number";
+            } else {
+                smallerDownloadOptionSuggestionMessage = "section";
+            }
```

Comment:
![image](https://cloud.githubusercontent.com/assets/16754650/23920018/096f6c9a-0934-11e7-9dd0-872f08fc9cb7.png)

How about this?